cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/PropertiesResolver.java

```diff
@@ -62,8 +62,7 @@
     systemArgs.put(Constants.CLUSTER_NAME, cConf.get(Constants.CLUSTER_NAME, ""));
     systemArgs.put(Constants.AppFabric.APP_SCHEDULER_QUEUE, queueResolver.getQueue(id.getNamespace()));
     if (SecurityUtil.isKerberosEnabled(cConf)) {
-      ImpersonationInfo impersonationInfo = new ImpersonationInfo(
-        namespaceQueryAdmin.get(id.toEntityId().getNamespaceId()), cConf);
+      ImpersonationInfo impersonationInfo = new ImpersonationInfo(id.getApplicationId(), cConf);
```

Comment:
This seems to be wrong. You are passing ApplicationId when the constructor needs a principal for which it should create ImpersonationInfo. The earlier code was querying namespaceQueryAdmin and passing the meta for creation.